### PR TITLE
Add CI Image for Dashboards Build

### PR DESCRIPTION
**Pull request by peternied**
### Description
I've copied the existing dashboard build image into a fresh file so these can diverge and included the `DOCKER_BUILD_SCRIPT_WITH_COMMANDS` in the image itself to keep the tagging consistent as changes are made.

See the image at https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&ordering=last_updated&name=ci-runner-centos7-opensearch-dashboards-build

Signed-off-by: Peter Nied <petern@amazon.com>
 
### Issues Resolved
- Follows up #1350 that configured the image
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kavilla — APPROVED** [review]

**Review by boktorbb-amzn — APPROVED** [review]
Looks good. One question, why is the node version being removed from the image name?

**Comment by peternied** [comment]
> Looks good. One question, why is the node version being removed from the image name?

When we are attempting to 'manage' all of the docker images, they all begin to blur together, while the name I've created is maybe not the best name, its at least easier to understand its intent and what would potentially break if we updated it

File: docker/ci/dockerfiles/ci-runner.centos7.opensearch-dashboards.build.dockerfile [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            MAJOR=${BASH_REMATCH[1]}; \
+            curl -LfsSo /tmp/openjdk-${MAJOR}.tar.gz ${BINARY_URL}; \
+            echo "${ESUM} */tmp/openjdk-${MAJOR}.tar.gz" | sha256sum -c -; \
+            mkdir -p /opt/java/openjdk-${MAJOR}; \
+            cd /opt/java/openjdk-${MAJOR}; \
+            tar -xf /tmp/openjdk-${MAJOR}.tar.gz --strip-components=1; \
+            rm -rf /tmp/openjdk-${MAJOR}.tar.gz; \
+            echo "export JAVA${MAJOR}_HOME=/opt/java/openjdk-${MAJOR}" >> /etc/profile.d/java_home.sh; \
+        fi; \
+    done;
```
Comment (peterzhuamazon) [inline docker/ci/dockerfiles/ci-runner.centos7.opensearch-dashboards.build.dockerfile:90]: There is no reason to include these JDKs in dashboards building.

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]
I have questions on this approach as it includes everything just like the main image.
Which to me is just duplications.

I would rather this image is simple and clean with only dashboards related libs installed.

Thoughts?

**Comment by peternied** [comment]
> I have questions on this approach as it includes everything just like the main image.

As we have a functional system today and limited automating testing, I'd like pair down what is included iteratively

File: docker/ci/dockerfiles/ci-runner.centos7.opensearch-dashboards.build.dockerfile (hunk 1 above)
Comment (peternied) [reply docker/ci/dockerfiles/ci-runner.centos7.opensearch-dashboards.build.dockerfile:90]: Note; I'd prefer to do this as a separate PR

Comment (boktorbb-amzn) [reply docker/ci/dockerfiles/ci-runner.centos7.opensearch-dashboards.build.dockerfile:90]: I can confirm that removing them shouldn't affect us @peternied . Will defer to you for whether it should be done in this PR or opening a separate one

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**